File: Java/civitas/src/civitas/CasillaCalle.java
package civitas;

import java.util.ArrayList;

//@author JuanCarlos Gonzalez Quesada
//EXAMEN

public class CasillaCalle extends Casilla{
    private TituloPropiedad titulo;
    private float importe;
    
    CasillaCalle(TituloPropiedad _titulo){
        super(_titulo.getNombre());
        titulo= _titulo;
        importe=_titulo.getPrecioCompra();
    }
        
    @Override
    public void recibeJugador(int iActual, ArrayList<Jugador> todos){
        System.out.println("Estoy en recibe jugador de CasillaCalle");
        if(super.jugadorCorrecto(iActual, todos)){
            super.informe(iActual, todos);
            if(!titulo.tienePropietario()){
                todos.get(iActual).puedeComprarCasilla();
                todos.get(iActual).setPuedeComprar(true);
            }
            else{
                titulo.tramitarAlquiler(todos.get(iActual));
                todos.get(iActual).setPuedeComprar(false);
            }
        }
    }
    
    public TituloPropiedad getTituloPropiedad(){
        return titulo;
    }
    
    @Override
    public String toString(){
        return "\nHas caido en " + super.getNombre() +"\n El importe es: " + Float.toString(importe) ;
    }
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/casilla_calle.rb
#encoding: utf­8
require_relative 'jugador'
require_relative "casilla.rb"

module Civitas
  class CasillaCalle < Casilla
    attr_accessor :titulo
    def initialize(_titulo)
      super(_titulo.nombre)
      @titulo = _titulo
      @importe = _titulo.precio_compra
    end
  
def recibe_jugador(iactual,todos)
  puts "Estoy en recibe_Jugador"
      if(jugador_correcto(iactual, todos))
        puts "Soy jugador correcto"
        informe(iactual, todos)
        if(@titulo.tiene_propietario()==true)
          puts "Titulo con propietario"
          if(@titulo.propietario!=todos.at(iactual)and @titulo.hipotecado==false and @titulo.propietario.encarcelado==false)
            puts "Titulo con propietario y pagas alquiler"
            todos.at(iactual).set_puede_comprar(false)
            todos.at(iactual).paga_alquiler(@titulo.alquiler_base)
            @titulo.propietario.recibe(@titulo.alquiler_base)
          else
            todos.at(iactual).set_puede_comprar(false)
            puts "Titulo con propietario y no pagas alquiler"
          end
        else
          puts "Titulo sin propietario"
          todos.at(iactual).set_puede_comprar(true)
        end
      end
end
    
  def to_sc()
    return "La casilla es " + nombre.to_s + " y el importe es " + @importe.to_s()
  end
end

  end
